db/db_impl.cc

```diff
@@ -3828,6 +3828,8 @@ SuperVersion* DBImpl::InstallSuperVersionAndScheduleWork(
 
   // Whenever we install new SuperVersion, we might need to issue new flushes or
   // compactions.
+  cfd->current()->storage_info()->ComputeCompactionScore(*cfd->ioptions(),
+                                                         mutable_cf_options);
```

Comment:
You may be able to call VersionSet::LogAndApply() with an empty VersionEdit. I'm not 100% sure though.